prow/cmd/deck/main.go

```diff
@@ -316,16 +317,21 @@ type pjClient interface {
 	GetProwJob(string) (kube.ProwJob, error)
 }
 
-func handleRerun(kc pjClient) http.HandlerFunc {
+func handleRerun(config configAgent, kc pjClient) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		name := r.URL.Query().Get("prowjob")
 		if !objReg.MatchString(name) {
 			http.Error(w, "Invalid ProwJob query", http.StatusBadRequest)
 			return
 		}
 		pj, err := kc.GetProwJob(name)
```

Comment:
FWIW this probably still has a timing attack but:
- the pod names are a bit long
- knowing the pod names shouldn't matter if we are sure to filter out access to them everywhere